### ISSUE TITLE
LogFormat breaks documented log formatting changes

### ISSUE THREAD

**Issue by crispywalrus** [twitter/twitter-server#47]
A Twitter-Server with LogFormat trait doesn't honor `defaultFormatter` overrides
### Expected behavior

`override def defaultFormatter = myFormatter` should change log format
### Actual behavior

the log format remains unchanged
### Steps to reproduce the behavior

Add the log formatter from the user guide and notice the unchanging log format. Removing LogFormat trait from the stack (by extending App and all the other traits rather than TwitterServer) fixes this and the override works as expected.

**Comment by crispywalrus** [comment]
To be complete about this, there is a difference between TwitterServer with and without LogFormat. This is the default format for log messages changes from
```
I 1020 21:03:17.630 THREAD179:  => com.twitter.server.handler.AdminRedirectHandler
I 1020 21:03:17.631 THREAD179: /admin/server_info => com.twitter.finagle.Filter$$anon$1
I 1020 21:03:17.631 THREAD179: /admin => com.twitter.server.handler.SummaryHandler
I 1020 21:03:17.632 THREAD179: /admin/contention => com.twitter.finagle.Filter$$anon$1
```
to
```
INF [20161020-13:58:03.104] services:  => com.twitter.server.handler.AdminRedirectHandler
INF [20161020-13:58:03.106] services: /admin => com.twitter.server.handler.SummaryHandler
INF [20161020-13:58:03.108] services: /admin/server_info => com.twitter.finagle.Filter$$anon$1
INF [20161020-13:58:03.110] services: /admin/contention => com.twitter.finagle.Filter$$anon$1
INF [20161020-13:58:03.110] services: /admin/lint => com.twitter.server.handler.LintHandler
INF [20161020-13:58:03.110] services: /admin/lint.json => com.twitter.server.handler.LintHandler
```

**Comment by ryanoneill** [comment]
Thank you for the report @crispywalrus. Would you be interested in submitting a pull request to fix it?

**Comment by cacoco** [comment]
@crispywalrus this should be fixed in with the changes in 906ae7df2d32e329ba12ec540b45274898cf5862. Since TwitterServer now uses SLF4J, you may want to take a look at the backwards compatibility section about logging: https://twitter.github.io/twitter-server/Features.html#backwards-compatibility-via-slf4j-jdk14.

Please let us know if you have any problems.

Thanks!
